libavcodec/hevcdec.c

```diff
@@ -729,7 +730,7 @@ static int hls_slice_header(HEVCContext *s)
             sh->rpl_modification_flag[0] = 0;
             sh->rpl_modification_flag[1] = 0;
             nb_refs = ff_hevc_frame_nb_refs(s);
-            if (!nb_refs) {
+            if (!nb_refs && !s->ps.pps->pps_curr_pic_ref_enabled_flag) {
```

Comment:
why we need !s->ps.pps->pps_curr_pic_ref_enabled_flag ?
if s->ps.pps->pps_curr_pic_ref_enabled_flag is true, we have ret++ in ff_hevc_frame_nb_refs. nb_refs never false.

Reply:
Fixed locally, thx.